test/models/vehicle_test.rb

```diff
@@ -25,19 +25,52 @@ def test_work_duration
       vrp = VRP.scheduling_seq_timewindows
       vrp = TestHelper.create(vrp)
 
-      assert_nil vrp.vehicles.first.work_duration
+      # work duration is only supposed to be called when there is no schedule
+      # therefore, work_duration should not be called if vehicle has sequence_timewindows
+      assert_raises OptimizerWrapper::DiscordantProblemError do
+        vrp.vehicles.first.work_duration
+      end
 
       vrp.vehicles.first.sequence_timewindows = []
       assert_equal 2**32, vrp.vehicles.first.work_duration
 
       vrp.vehicles.first.timewindow = { start: 10 }
-      assert_nil vrp.vehicles.first.work_duration
+      assert_equal 2**32, vrp.vehicles.first.work_duration
 
       vrp.vehicles.first.timewindow = { end: 10 }
-      assert_nil vrp.vehicles.first.work_duration
+      assert_equal 2**32, vrp.vehicles.first.work_duration
 
       vrp.vehicles.first.timewindow = { start: 10, end: 20 }
       assert_equal 10, vrp.vehicles.first.work_duration
+
+      vrp.vehicles.first.duration = 5
+      assert_equal 5, vrp.vehicles.first.work_duration
+    end
+
+    def test_total_work_time_in_range
+      vrp = VRP.lat_lon_scheduling
+      vrp = TestHelper.create(vrp)
+      assert_equal 40, vrp.vehicles.first.total_work_time_in_range(0, 3)
+
+      vrp.vehicles.first.ignore_computed_data
+      vrp.vehicles.first.duration = 5
+      assert_equal 20, vrp.vehicles.first.total_work_time_in_range(0, 3)
+
+      vrp.vehicles.first.ignore_computed_data
```

Comment:
The method should be named _reset_computed_data_. It seems more meaningful to me